score.py

```diff
@@ -128,6 +134,75 @@ def run_inference(images_with_header: List[ImageWithHeader],
     return segmentation
 
 
+def extract_zipped_dicom_series(zip_file_path: Path, model_folder: Path) -> List[Path]:
+    """
+    Unzip a zip file. If it contains multiple directories, assume each is a DICOM series.
+    Otherwise assume that it is a single DICOM series.
+    Return a list of DICOM series folders.
+
+    :param zip_file_path: Path to zip file.
+    :param model_folder: Path to model folder.
+    :return: List of DICOM series folders.
+    """
+    dicom_folders: List[Path] = []
+    extraction_folder = model_folder / "temp_extraction"
+    with zipfile.ZipFile(zip_file_path, 'r') as zip_file:
+        for zipped_file in zip_file.namelist():
+            if zipped_file.endswith('.dcm'):
+                zip_file.extract(zipped_file, str(extraction_folder))
+                extracted_file_path = extraction_folder / zipped_file
+                extracted_file_path_folder = extracted_file_path.parent
+                if extracted_file_path_folder not in dicom_folders:
+                    dicom_folders.append(extracted_file_path_folder)
+    return dicom_folders
+
+
+def convert_zipped_dicom_to_nifti(test_images: List[Path], model_folder: Path) -> Tuple[List[Path], List[Path]]:
```

Comment:
I think this overcomplicated matters. It can be simplified to "Take any ZIP file. Extract it. Load everything in it assuming that it's a single dicom series". The worst thing that could happen is that the actual dicom files are in a folder structure like A/B/foo.dcm - that's something the code must handle. But nothing as complicated as multiple series in multiple folder. (the gateway does also not presently have such functionality)